src/Dictionaries/SSDComplexKeyCacheDictionary.cpp

```diff
@@ -1447,6 +1449,12 @@ void SSDComplexKeyCacheDictionary::getItemsNumberImpl(
     const Columns & key_columns, const DataTypes & key_types,
     ResultArrayType<OutputType> & out, DefaultGetter && get_default) const
 {
+    assert(dict_struct.key);
+    assert(key_columns.size() == key_types.size());
+    assert(key_columns.size() == dict_struct.key->size());
+
+    dict_struct.validateKeyTypes(key_types);
```

Comment:
Only two calls to `validateKeyTypes` matter.
Everything else is just "defensive programming".